pkg/action/executor/machine_scalers.go

```diff
@@ -130,9 +131,32 @@ func (controller *machineSetController) checkPreconditions() error {
 
 // executeAction scales a MachineSet by modifying its replica count
 func (controller *machineSetController) executeAction() error {
-	desiredReplicas := controller.machineSet.Status.Replicas + controller.request.diff
+	diff := controller.request.diff
+	client := controller.request.client
+	desiredReplicas := controller.machineSet.Status.Replicas + diff
 	controller.machineSet.Spec.Replicas = &desiredReplicas
-	machineSet, err := controller.request.client.machineSet.Update(controller.machineSet)
+
+	if diff < 0 {
```

Comment:
Just wanted to confirm: When diff is negative, is it always -1?  Meaning scaling down by 1?  If so, then all this makes sense.

Reply:
Yes, thats right. Its always -1.